### ISSUE TITLE
[feat] Option to disable top-k routing weights normalization

### ISSUE THREAD

**Issue by sohamparikh** [ServiceNow/Fast-LLM#83]
# 🧐 Problem Description

OLMoE has [disabled the normalization for top-k routing probabilities](https://huggingface.co/allenai/OLMoE-1B-7B-0924/blob/6d84c48581ece794365f2b8e9cfb043c68ade9c5/config.json#L15). There is no clear motivation or ablation for why this was done. [DeepSeekMoE](https://huggingface.co/deepseek-ai/deepseek-moe-16b-base/blob/521d2bc4fb69a3f3ae565310fcc3b65f97af2580/config.json#L25) also disables top-k normalization, while Mixtral-8x7b-v0.1 normalizes them. 

# 💡 Proposed Solution
Apply softmax before `torch.topk` in https://github.com/ServiceNow/Fast-LLM/blob/51d57158d625883da189bcce3af3c8908e527824/fast_llm/layers/transformer/mixture_of_experts.py#L167

# 🔄 Alternatives Considered
Normalize top-k scores as usual, since there's no clear motivation for the same. Good thing is that it's config driven in the HF implementation for OLMoE

# 📈 Potential Benefits
No clear benefits, but it could instead slow down training by a bit since now we're applying softmax on logits from all experts.

# 📝 Additional Context

See [OLMoE implementation](https://github.com/huggingface/transformers/blob/f9c7e6021e9a9a9fd3fc8bb291da9451066aeb8d/src/transformers/models/olmoe/modeling_olmoe.py#L646) for reference

**Comment by sohamparikh** [comment]
Reply from the OLMoE authors for reference. There might not be strong reasons to do this as of now

https://huggingface.co/allenai/OLMoE-1B-7B-0924/discussions/4#674e9e974232e14c900fa528
